autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/sync/CredentialsSync.kt

```diff
@@ -0,0 +1,186 @@
+    }
+
+    suspend fun getCredentialsForDomain(domain: String?): List<LoginCredentials> {
+        return secureStorage.websiteLoginDetailsWithCredentialsForDomain(domain.orEmpty()).firstOrNull()?.map {
+            it.toLoginCredentials()
+        } ?: emptyList()
+    }
+
+    suspend fun saveCredential(
+        loginCredential: LoginCredentials,
+        remoteId: String,
+    ) {
+        val loginDetails = WebsiteLoginDetails(
+            id = loginCredential.id,
+            domain = loginCredential.domain,
+            username = loginCredential.username,
+            domainTitle = loginCredential.domainTitle,
+            lastUpdatedMillis = loginCredential.lastUpdatedMillis,
+        )
+        val webSiteLoginCredentials = WebsiteLoginDetailsWithCredentials(
+            details = loginDetails,
+            password = loginCredential.password,
+            notes = loginCredential.notes,
+        )
+
+        secureStorage.addWebsiteLoginDetailsWithCredentials(webSiteLoginCredentials)?.details?.id?.let { autofillId ->
+            credentialsSyncMetadata.addOrUpdate(
+                CredentialsSyncMetadataEntity(syncId = remoteId, localId = autofillId, deleted_at = null, modified_at = null),
+            )
+        }
+    }
+
+    suspend fun updateCredentials(
+        loginCredential: LoginCredentials,
+        remoteId: String,
+    ) {
+        val loginDetails = WebsiteLoginDetails(
+            id = loginCredential.id,
+            domain = loginCredential.domain,
+            username = loginCredential.username,
+            domainTitle = loginCredential.domainTitle,
+            lastUpdatedMillis = loginCredential.lastUpdatedMillis,
+        )
+        val webSiteLoginCredentials = WebsiteLoginDetailsWithCredentials(
+            details = loginDetails,
+            password = loginCredential.password,
+            notes = loginCredential.notes,
+        )
+        secureStorage.updateWebsiteLoginDetailsWithCredentials(webSiteLoginCredentials)?.details?.id?.let { autofillId ->
+            credentialsSyncMetadata.addOrUpdate(
+                CredentialsSyncMetadataEntity(syncId = remoteId, localId = autofillId, deleted_at = null, modified_at = null),
+            )
+        }
+    }
+
+    suspend fun deleteCredential(localId: Long) {
+        secureStorage.deleteWebsiteLoginDetailsWithCredentials(localId)
+        credentialsSyncMetadata.removeEntityWith(localId)
+    }
+
+    private suspend fun allContentAsUpdates() = secureStorage.websiteLoginDetailsWithCredentials().firstOrNull().mapToLoginCredentialEntry()
+
+    private suspend fun changesSince(since: String): List<LoginCredentialEntry> {
+        Timber.d("CredentialsSync: generating changes since $since")
+        credentialsSyncMetadata.getAllCredentials().forEach {
+            Timber.i("CredentialsSync: syncMetadata $it")
+        }
+
+        val values2 = credentialsSyncMetadata.getChangesSince(since).map {
```

Comment:
is there a better name than `values2`?

Reply:
`values007` jk 

good catch, I can go for `changes` instead.